gammapy/data/data_store.py

```diff
@@ -416,6 +435,54 @@ def get_observations(
         log.info(f"Observations selected: {len(obs_list)} out of {len(obs_id)}.")
         return Observations(obs_list)
 
+    def get_observation_groups(self, key, **kwargs):
+        """Generate groups of `~gammapy.data.Observations` with a shared property.
+
+        Parameters
+        ----------
+        key : str
+            Key of the observation table used to apply the grouping.
+            For example, "EVENT_TYPE" will return group observations
+            with the same event type.
+        kwargs : dict, optional
+            Keyword arguments passed to `~gammapy.data.DataStore.get_observations`.
+
+        Returns
+        -------
+        groups : dict of `~gammapy.data.Observations`
+            Dictionary of Observations instance, one instance for each group.
+
+
+        Examples
+        --------
+        Create groups of observation with the same event type.
+
+        >>> from gammapy.data import DataStore
+        >>> path = "$GAMMAPY_DATA/tests/format/swgo/"
+        >>> datastore = DataStore.from_dir(path, "hdu-index.fits.gz", "obs-index.fits.gz")
+        >>> observation_groups = datastore.get_observation_groups("EVENT_TYPE")
+        """
+
+        if self.obs_table is None:
+            raise ValueError(
+                "obs_table attribute must not be None to select groups of observations"
+            )
+
+        observations = self.get_observations(**kwargs)
+        obs_table = self.obs_table[
```

Comment:
This can be simplified using directly `ObservationTable.select_obs_id([_ for _ in observations.ids])`.